api/v1beta1/foundationdbcluster_types.go

```diff
@@ -470,16 +470,45 @@ type ProcessGroupStatus struct {
 	// Addresses represents the list of addresses the process group has been known to have.
 	Addresses []string `json:"addresses,omitempty"`
 	// Remove defines if the process group is marked for removal.
+	// Deprecated: Use RemovalTimestamp instead.
 	Remove bool `json:"remove,omitempty"`
+	// RemoveTimestamp if not empty defines when the process group was marked for removal.
+	RemovalTimestamp *metav1.Time `json:"removalTimestamp,omitempty"`
 	// Excluded defines if the process group has been fully excluded.
 	// This is only used within the reconciliation process, and should not be considered authoritative.
+	// Deprecated: Use ExclusionTimestamp instead.
 	Excluded bool `json:"excluded,omitempty"`
+	// ExcludedTimestamp defines when the process group has been fully excluded.
+	// This is only used within the reconciliation process, and should not be considered authoritative.
+	ExclusionTimestamp *metav1.Time `json:"exclusionTimestamp,omitempty"`
```

Comment:
Thinking more about it I actually think we can remove the `exclude` state completely. We have a few places where we check for that state e.g. in the replace failed process group sub-reconciler which could also use the `ResourcesTerminating` state to see if the process group was already excluded.

Reply:
We're also using it to track metrics on exclusions, but maybe it's better to look at that through the database status.